### PR TITLE
Add new allowGettingAccessTokenWithRefreshToken in MSALSilentTokenPar…

### PR DESCRIPTION
**Pull request by kaisong1990**
…ameters public API

## Proposed changes

Add new property in MSALSilentTokenParameters public API to allow using local RT as fallback if broker fails

[CommonCore change can be found here](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1160)

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/src/public/MSALSilentTokenParameters.h [hunk 1]
```diff
@@ -42,6 +42,15 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic) BOOL forceRefresh;
 
+/**
+ 1. When broker is presenting on the device
+    Default is false. when broker failed to return a (new) access token, ignores existing refresh token in local cahce, and return broker error.
```
Comment (juan-arias) [inline MSAL/src/public/MSALSilentTokenParameters.h:47]: nit: typo
```suggestion
    Default is false. when broker failed to return a (new) access token, ignores existing refresh token in local cache, and return broker error.
```

**Review by juan-arias — APPROVED** [review]

File: MSAL/src/public/MSALSilentTokenParameters.h [hunk 2]
```diff
@@ -42,6 +42,15 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic) BOOL forceRefresh;
 
+/**
+ 1. When broker is presenting on the device
+    Default is false. when broker failed to return a (new) access token, ignores existing refresh token in local cache, and return broker error.
+    If set to true, when broker failed to return a (new) access token, tries with existing refresh token in the cache, and return results.
+ 2. When broker is not presenting on the device
```
Comment (antrix1989) [inline MSAL/src/public/MSALSilentTokenParameters.h:49]: How do we know that broker is available? Does this applies only to sso ext? If yes, should we replace the word "broker" with "SSO ext"?

Comment (kaisong1990) [reply MSAL/src/public/MSALSilentTokenParameters.h:49]: Yes, for silent flow, this refers to Sso Ext

Comment (kaisong1990) [reply MSAL/src/public/MSALSilentTokenParameters.h]: Updated

File: MSAL/src/public/MSALSilentTokenParameters.h [hunk 3]
```diff
@@ -42,6 +42,15 @@ NS_ASSUME_NONNULL_BEGIN
...
 @property (nonatomic) BOOL forceRefresh;
 
+/**
+ 1. When Sso Extension is presenting on the device
+    Default is false. when Sso Extension failed to return a (new) access token, ignores existing refresh token in local cahce, and return Sso Extension error.
+    If set to true, when Sso Extension failed to return a (new) access token, tries with existing refresh token in the cache, and return results.
+ 2. When Sso Extension is not presenting on the device
+    This parameter is ignored, and tries with existing refresh token in the cache.
+ */
+@property (nonatomic) BOOL allowGettingAccessTokenWithRefreshToken;
```
Comment (antrix1989) [inline MSAL/src/public/MSALSilentTokenParameters.h:52]: This name is confusing. Without reading the comment above, there is no hint this logic applies only when SSO ext is available.

Comment (kaisong1990) [reply MSAL/src/public/MSALSilentTokenParameters.h]: Updated the same

**Review by oldalton — APPROVED** [review]

**Pull request merged**